frontend/src/metabase/writeback/components/ActionCreator/InlineDataReference.styled.tsx

```diff
@@ -0,0 +1,32 @@
+import styled from "@emotion/styled";
+import { color } from "metabase/lib/colors";
+
+import Button from "metabase/core/components/Button";
+
+export const DataReferenceContainer = styled.div<{ isOpen: boolean }>`
+  flex: ${props => (props.isOpen ? 1 : 0)} 1 0;
+  overflow: ${props => (props.isOpen ? "auto" : "hidden")};
+  position: relative;
+  height: 100%;
+  background-color: ${color("white")};
+  border-left: 1px solid ${color("border")};
+  border-right: 1px solid ${color("border")};
+  transition: flex-grow 500ms ease-in-out,
+    opacity 250ms ease-in-out ${props => (props.isOpen ? "250ms" : "")};
+  opacity: ${props => (props.isOpen ? 1 : 0)};
```

Comment:
For me, 500ms felt quite slow when I was playing with the editor locally. Is there any reason not to make the open/close experience instant as we do in the native query editor?

Reply:
i tried it instant, and it felt a little jarring without some transition in the middle of the page, we can go faster though!